indexer.ts

```diff
@@ -0,0 +1,743 @@
+    "chef_set_payouts",
+  ];
+
+  async function handleNewPot(
+    args,
+    receiverId,
+    signerId,
+    predecessorId,
+    receiptId
+  ) {
+    let data = JSON.parse(args);
+    console.log("new pot data::", { ...data }, receiverId, signerId);
+    await context.db.Account.upsert({ id: data.owner }, ["id"], []);
+    await context.db.Account.upsert({ id: signerId }, ["id"], []);
+
+    if (data.chef) {
+      await context.db.Account.upsert({ id: data.chef }, ["id"], []);
+    }
+
+    if (data.admins) {
+      for (const admin in data.admins) {
+        await context.db.Account.upsert({ id: admin }, ["id"], []);
+        let pot_admin = {
+          pot_id: receiverId,
+          admin_id: admin,
+        };
+        await context.db.PotAdmin.insert(pot_admin);
+      }
+    }
+
+    const deploy_time = new Date(
+      Number(BigInt(block.header().timestampNanosec) / BigInt(1000000))
+    );
+
+    const potObject = {
+      id: receiverId,
+      pot_factory_id: predecessorId,
+      deployer_id: signerId,
+      deployed_at: deploy_time,
+      source_metadata: JSON.stringify(data.source_metadata),
+      owner_id: data.owner,
+      chef_id: data.chef,
+      name: data.pot_name,
+      description: data.pot_description,
+      max_approved_applicants: data.max_projects,
+      base_currency: null,
+      application_start: new Date(data.application_start_ms),
+      application_end: new Date(data.application_end_ms),
+      matching_round_start: new Date(data.public_round_start_ms),
+      matching_round_end: new Date(data.public_round_end_ms),
+      registry_provider: data.registry_provider,
+      min_matching_pool_donation_amount: data.min_matching_pool_donation_amount,
+      sybil_wrapper_provider: data.sybil_wrapper_provider,
+      custom_sybil_checks: data.custom_sybil_checks,
+      custom_min_threshold_score: data.custom_min_threshold_score,
+      referral_fee_matching_pool_basis_points:
+        data.referral_fee_matching_pool_basis_points,
+      referral_fee_public_round_basis_points:
+        data.referral_fee_public_round_basis_points,
+      chef_fee_basis_points: data.chef_fee_basis_points,
+      total_matching_pool: data.total_matching_pool || "0",
+      total_matching_pool_usd: data.total_matching_pool_usd,
+      matching_pool_balance: data.matching_pool_balance || "0",
+      matching_pool_donations_count: data.matching_pool_donations_count || 0,
+      total_public_donations: data.total_public_donations || "0",
+      total_public_donations_usd: data.total_public_donations_usd,
+      public_donations_count: data.public_donations_count || 0,
+      cooldown_end: data.cooldown_end,
+      all_paid_out: false,
+      protocol_config_provider: data.protocol_config_provider,
+    };
+
+    await context.db.Pot.insert(potObject);
+
+    let activity = {
+      signer_id: signerId,
+      receiver_id: receiverId,
+      timestamp: deploy_time,
+      type: "Deploy_Pot",
+      action_result: potObject.id,
```

Comment:
not sure how useful this is. wasn't the idea to store the decoded success value in this field? otherwise perhaps a raw dump of the action or something... open to ideas

Reply:
nevermind, i understand what you're doing here now, makes sense